source/arm7/ieee_802_11/beacon.c

```diff
@@ -415,17 +415,25 @@ void Wifi_ProcessBeaconOrProbeResponse(Wifi_RxHeader *packetheader, int macbase)
                 // only use RSSI when we're on the right channel
                 int tmp = packetheader->rssi_ & 255;
 
-                for (int j = 0; j < 7; j++)
+                for (int j = 0; j < 8; j++)
                     ap->rssi_past[j] = tmp;
             }
             else
             {
                 // Update RSSI later.
-                for (int j = 0; j < 7; j++)
+                for (int j = 0; j < 8; j++)
                     ap->rssi_past[j] = 0;
             }
         }
 
+        // If the WifiData AP MAC is the same as this beacon MAC, then update
+        // RSSI in WifiData as well.
+        if (Wifi_CmpMacAddr(WifiData->apmac7, data + HDR_MGT_SA))
+        {
+            for (int j = 0; j < 8; j++)
+                WifiData->rssi_past[j] = ap->rssi_past[j];
```

Comment:
I think I would just add the 8 values and get the mean here, and save it to a `u8 rssi` in the shared struct instead of an array, that way we save a bit of shared RAM. And the code here is basically the same.